File: CareConnect/CareConnect.Models/Requests/ClientUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;
using CareConnect.Models.Requests;
using CareConnect.Models.Responses;

namespace CareConnect.Models.Requests
{
    public class ClientUpdateRequest
    {
        public bool? EmploymentStatus { get; set; }

        [JsonIgnore]
        public DateTime? ModifiedDate { get; set; } = DateTime.Now;

        public UserUpdateRequest? User { get; set; }
    }
}

File: UI/careconnect_admin/lib/models/requests/client_update_request.dart
import 'package:careconnect_admin/models/requests/user_update_request.dart';

import 'package:json_annotation/json_annotation.dart';

part 'client_update_request.g.dart'; // Don't open or edit this

@JsonSerializable()
class ClientUpdateRequest {
  final bool? employmentStatus;
  final UserUpdateRequest user;

  ClientUpdateRequest({required this.employmentStatus, required this.user});

  factory ClientUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$ClientUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ClientUpdateRequestToJson(this);
}
